doc/HOWTO/howto_readState.md

```diff
@@ -0,0 +1,164 @@
+k_foo = getLedgerKeySymbol(contractID, "FOO")
+k_bar = getLedgerKeySymbol(contractID, "BAR")
+```
+
+With the key, or set of keys, in hand, we now invoke the RPC, to querry the current values of the fields these keys represent.
+
+In Javascript, querying the RPC is done as follows:
+```js
+let res = await fetch(NET, {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json',
+  },
+  body: JSON.stringify(requestBody),
+})
+let json = await res.json()
+```
+where `NET` is the Soroban network being queried, for example 
+`'https://soroban-testnet.stellar.org'`, and `requestBody` is the specific request we're making of the network.
+
+Since we aim to read a ledger entry, our request will use the `getLedgerEntries` method, and `requestBody` will look like this:
+```js
+let requestBody = {
+  "jsonrpc": "2.0",
+  "id": ...,
+  "method": "getLedgerEntries",
+  "params": {
+    "keys": [
+      k
+    ]
+  }
+}
+```
+
+Here, if we had multiple fields to read, we would simply use
+```js
+...
+"keys": [
+      k, k_foo, k_bar
+    ]
+```
+
+Our result would look like this:
+```js
+{
+  "jsonrpc": "2.0",
+  "id": ...,
+  "result": {
+    "entries": [
+      {
+        "key": ...,
+        "xdr": XDR,
+        "lastModifiedLedgerSeq": ...
+      }
+    ],
+    "latestLedger": ...
+  }
+}
+```
+
+where the value of the field that we seek is contained within `XDR`. With multiple keys, the difference would simply be
+```js
+...
+"entries": [
+      {
+        "key": ...,
+        "xdr": XDR,
+        "lastModifiedLedgerSeq": ...
+      },
+      {
+        "key": ...,
+        "xdr": XDR_FOO,
+        "lastModifiedLedgerSeq": ...
+      }
+      {
+        "key": ...,
+        "xdr": XDR_BAR,
+        "lastModifiedLedgerSeq": ...
+      }
+    ],
+```
+
+The final step is unpacking the obtained XDR, from which we  read the value associated with our key. This [functionality][sdk] is available from the `stellar-sdk` (`LedgerEntryData.from_xdr`), as well as [online][lab] for quick experimentation.
+
+Reading the `XDR` value, we should end up with the following shape:
+```
+LedgerEntryData: [contractData]
+    contractData
+        ext: [undefined]
+        contract: [scAddressTypeContract]
+            contractId: ...
+        key: [scvVec]
+            vec: Array[2]
+                [0]: [scvSymbol]
+                    sym: Counter
+                [1]: [scvAddress]
+                    address: ...
+        durability: {"name":"persistent","value":1}
+        val: [scvU32]
+            u32: VAL
+```
+Where `VAL` is the value we need to read for `COUNTER`.
+If the field we're interested in was some other type, say a map, it would look like this:
+```
+...
+val: [scvMap]
+    map: { "foo": 100, "bar": true, ...}
+```
+
+To summarize, these are the steps required to create a State:
+0. Obtain a `contractId`.
+1. Identify a collection of field identifiers (`"FOO", "BAR", "BAZ", ...`) that will make up the State
```

Comment:
Is there a programmatic way to access contract metadata such as the field names and their types?

Reply:
There's a way to get the entire contract bytecode, so I would have to imagine yes. Needs more looking into though.

Reply:
shall we expect more HOWTOs on the way?

Reply:
Ideally one more for tx data, and then a longer one for setting up a testing environment, see #6. 
I imagine this particular question will be resolved in the last part, once I have the local instances set up and can actually fiddle with inputs.